src/main/java/com/cloudbees/jenkins/support/SupportAction.java

```diff
@@ -114,6 +130,72 @@ public Localizable getActionBlurb() {
         return SupportPlugin.getComponents();
     }
 
+    public class Bundle {
+        File location;
+        public Bundle(File location) {
+            this.location = location;
+        }
+
+        public File getFile() {
+            return location;
+        }
+
+        public String getName() {
+            return location.getName();
+        }
+
+        public String getHref() {
+            return "bundles/bundle/" + getName();
+        }
+
+        public long getLength() {
+            return location.length();
+        }
+
+        private void doDownload(StaplerRequest req, StaplerResponse rsp) throws IOException {
+            OutputStream os = rsp.getOutputStream();
+            try {
+                rsp.setContentType("application/zip");
+                Util.copyStreamAndClose(new FileInputStream(location), os);
+            } finally {
+                IOUtils.closeQuietly(os);
+            }
+        }
+    }
+
+    public final class BundleList extends ArrayList<Bundle> {
+        public void doBundle(StaplerRequest req, StaplerResponse rsp) throws IOException {
+            String restOfPath = req.getRestOfPath();
```

Comment:
Permission check - bundle may contain information that an arbitrary user should not be able to see?